src/StatsigOnPrem.ts

```diff
@@ -150,30 +176,60 @@ export default class StatsigOnPrem implements StatsigInterface {
       name,
       salt: IDUtils.generateNewSalt(),
       idType: idType ?? "userID",
+      targetApps: new Set(targetApps ?? []),
       ...metadata,
     };
-    await this.store.addGate(name, gate, targetApps);
-    await this.cache.clearAll();
+    await this.store.addGate(gate);
+    await this.clearCacheForEntity(gate);
   }
 
   public async updateGate(
     name: string,
     args: Partial<FeatureGateCreationArgs>
   ): Promise<void> {
-    const { targetApps, ...metadata } = args;
-    const existing = await this.getGate(name);
-    if (existing == null) {
+    const gate = await this.getGate(name);
+    if (gate == null) {
       console.warn("Attempting to update non-existent gate");
       return;
     }
-    const updatedGate: FeatureGate = { ...existing, ...metadata };
-    await this.store.updateGate(name, updatedGate, targetApps);
-    await this.cache.clearAll();
+    await this.store.updateGate(gate, args);
+    await this.clearCacheForEntity(gate);
```

Comment:
`targetApps` can be modified by this method, so I think this might also need to clear the cache for the superset of `existingGate.targetApps + updatedGate.targetApps`

Reply:
Ah yes good point!